src/modules/sensors/vehicle_gps_position/VehicleGPSPosition.cpp

```diff
@@ -490,86 +508,83 @@ void VehicleGPSPosition::update_gps_blend_states()
-	double lat_deg_now = (double)_gps_blended_state.lat * 1.0e-7;
-	double lon_deg_now = (double)_gps_blended_state.lon * 1.0e-7;
-	double lat_deg_res, lon_deg_res;
-	add_vector_to_global_position(lat_deg_now, lon_deg_now, blended_NE_offset_m(0), blended_NE_offset_m(1), &lat_deg_res,
-				      &lon_deg_res);
-	_gps_blended_state.lat = (int32_t)(1.0E7 * lat_deg_res);
-	_gps_blended_state.lon = (int32_t)(1.0E7 * lon_deg_res);
-	_gps_blended_state.alt += (int32_t)blended_alt_offset_mm;
+	const double lat_deg_now = (double)gps_blended_state.lat * 1.0e-7;
+	const double lon_deg_now = (double)gps_blended_state.lon * 1.0e-7;
+	double lat_deg_res = 0;
+	double lon_deg_res = 0;
+	add_vector_to_global_position(lat_deg_now, lon_deg_now,
+				      blended_NE_offset_m(0), blended_NE_offset_m(1),
+				      &lat_deg_res, &lon_deg_res);
+	gps_blended_state.lat = (int32_t)(1.0E7 * lat_deg_res);
+	gps_blended_state.lon = (int32_t)(1.0E7 * lon_deg_res);
+	gps_blended_state.alt += (int32_t)blended_alt_offset_mm;
 
 	// Take GPS heading from the highest weighted receiver that is publishing a valid .heading value
 	uint8_t gps_best_yaw_index = 0;
-	best_weight = 0.0f;
+	float best_yaw_weight = 0.0f;
 
 	for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
-		if (PX4_ISFINITE(_gps_state[i].heading) && (_blend_weights[i] > best_weight)) {
-			best_weight = _blend_weights[i];
+		if (PX4_ISFINITE(_gps_state[i].heading) && (blend_weights[i] > best_yaw_weight)) {
+			best_yaw_weight = blend_weights[i];
 			gps_best_yaw_index = i;
 		}
 	}
 
-	_gps_blended_state.heading = _gps_state[gps_best_yaw_index].heading;
-	_gps_blended_state.heading_offset = _gps_state[gps_best_yaw_index].heading_offset;
+	gps_blended_state.heading = _gps_state[gps_best_yaw_index].heading;
+	gps_blended_state.heading_offset = _gps_state[gps_best_yaw_index].heading_offset;
+
+	return gps_blended_state;
 }
 
-void VehicleGPSPosition::update_gps_offsets()
+void VehicleGPSPosition::update_gps_offsets(const sensor_gps_s &gps_blended_state)
 {
 	// Calculate filter coefficients to be applied to the offsets for each GPS position and height offset
-	// Increase the filter time constant proportional to the inverse of the weighting
 	// A weighting of 1 will make the offset adjust the slowest, a weighting of 0 will make it adjust with zero filtering
-	float alpha[GPS_MAX_RECEIVERS] = {};
+	float alpha[GPS_MAX_RECEIVERS] {};
 	float omega_lpf = 1.0f / fmaxf(_param_sens_gps_tau.get(), 1.0f);
 
 	for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
 		if (_gps_state[i].timestamp - _time_prev_us[i] > 0) {
 			// calculate the filter coefficient that achieves the time constant specified by the user adjustable parameter
-			float min_alpha = constrain(omega_lpf * 1e-6f * (float)(_gps_state[i].timestamp - _time_prev_us[i]),
-						    0.0f, 1.0f);
-
-			// scale the filter coefficient so that time constant is inversely proprtional to weighting
```

Comment:
isn't that useful?

Reply:
It looks like it was removed back in ekf2, but somehow got lost in the drawn out PR sitting around for 6 months.

https://github.com/PX4/PX4-Autopilot/commit/8de1f5229bd70cccaa8eb2605fbfb6a17b304176#diff-72bb106f8fdfef0de0e8174f5b63ad2987bd47a8f07ff445bde40a812c87651e